### ISSUE TITLE
Force good settings for known batteries

### ISSUE THREAD

**Issue by izak** [victronenergy/venus#511]
When a BYD battery is detected, apply the following settings:

DVCC = on
SVS = off
STS = off

- [X] Extend the possible values for these settings, so that forced settings can be carried in the higher bits:

| bit 1 | bit 0 | Meaning
|-------|-------|--------------:|
|      0 |     0 |             Off |
|      0 |     1 |             On |
|      1 |     0 |  Forced off |
|      1 |     1 |  Forced on |

This translates simply to values 0-3.

- [x] Extend the GUI display to display the correct value and make it read-only when the value is forced.

The purpose of this change is to make it easier for installers and support people.

**Comment by mpvader** [comment]
MG electronics: DVCC force on, SVS force off, STS force off

**Comment by izak** [comment]
There are some corner cases here.

We want to automatically force the setting (and lock/hide it in the GUI) if we know a good set of values for it.

* What if there is more than one battery in the system of different types? Answer: Only do this for the selected battery monitor.
* And if the battery monitor changes? Answer: Then unlock the setting and leave it where it was.

Therefore the locking/unlocking code fires on: 1) A new battery is detected, 2) it's a battery we have known-good settings for, 3) this battery is the battery monitor (either auto- or manually selected).

For this to work well, a way is needed for the delegates to see when the battery monitor changes. This does not exist yet.

**Comment by izak** [comment]
Some GUI decisions. When DVCC is locked on, remove the firmware notice. It is pointless to warn the user over something he has no control over.

![Selection_350](https://user-images.githubusercontent.com/851549/65038463-4e8d3980-d950-11e9-8423-754cf1015a13.png)

Forced settings looks like this:

![Selection_349](https://user-images.githubusercontent.com/851549/65038507-649afa00-d950-11e9-886e-f4f7934c209a.png)

**Comment by mpvader** [comment]
This links to another issue we have open.: raise a warning about too old firmware; there is an issue for that.

I hadn’t thought about this one; it sounds like now the moment has come that we need to make that warning. Do it in systemcalc so that wanring is available both in gui & on vrm?

**Comment by mpvader** [comment]
@izak you wrote:

> For this to work well, a way is needed for the delegates to see when the battery monitor changes. This does not exist yet.

Why not evaluate this on each iteration in the main loop?

Then you’ll be sure to never miss an event.

Check active main battery type against the actual values of the dvcc & svs & sts settings.

**Comment by izak** [comment]
I found a reliable spot to hook this so that we don't have to do the entire (unnecessary) calculation on every loop.

Systemcalc already calculates the active battery monitor every time and updates dbus only if it changes:

https://github.com/victronenergy/dbus-systemcalc-py/blob/master/dbus_systemcalc.py#L337

So I extended this to delegates.

The forced settings are applied at startup or when the battery is added too.

https://github.com/victronenergy/dbus-systemcalc-py/blob/master/dbus_systemcalc.py#L337
